source/RoboticsClub/Robot/robot_control-fall_2017.c

```diff
@@ -0,0 +1,165 @@
+
+void updateWheels();
+void updateClaw();
+void updateArm();
+void updateForklift();
+void updateLaunch();
+
+//+++++++++++++++++++++++++++++++++++++++++++++| MAIN |+++++++++++++++++++++++++++++++++++++++++++++++
+task main ()
+{
+    int tick = 0;
+
+    int update_tick = 0;
+    const int update_ms = 5;
+
+    while(true)
+    {
+        tick = time1[T1];
+
+        if (tick - update_tick >= update_ms)
+        {
+            update_tick = tick;
+
+            updateWheels();
+
+            updateClaw();
+            updateArm();
+
+
+            updateLaunch();
+            updateForklift();
+        }
+    }
+}
+
+void updateWheels()
+{
+    const int threshold = 10;   // helps to eliminate 'noise' from a joystick that isn't perfectly at (0,0)
+
+    int joy_x = vexRT[Ch1];   // This is the RIGHT analog stick.  For LEFT, change 'Ch1' to 'Ch4'.
+    int joy_y = vexRT[Ch2];   // This is the RIGHT analog stick.  For LEFT, change 'Ch2' to 'Ch3'.
+
+    // Forward, and swing turns: (both abs(X) and abs(Y) are above the threshold, and Y is POSITIVE)
+    if((abs(joy_x) > threshold) && (abs(joy_y) > threshold) && (joy_y > 0))
+    {
+        motor[leftMotor]  = (joy_y + joy_x)/4;
+        motor[rightMotor] = (joy_y - joy_x)/4;
+    }
+    // Backwards and swing turns: (both abs(X) and abs(Y) are above the threshold, and Y is NEGATIVE)
+    else if((abs(joy_x) > threshold) && (abs(joy_y) > threshold) && (joy_y < 0))
+    {
+        motor[leftMotor]  = (joy_y - joy_x)/4;
+        motor[rightMotor] = (joy_y + joy_x)/4;
+    }
+
+    // Turning in place: (abs(X) is above the threshold, abs(Y) is below the threshold)
+    else if((abs(joy_x) > threshold) && (abs(joy_y) < threshold))
+    {
+        motor[leftMotor]  = joy_x;
+        motor[rightMotor] = (-1 * joy_x);
+    }
+    // Standing still: (both abs(X) and abs(Y) are below the threshold)
+    else
+    {
+        motor[leftMotor]  = 0;
+        motor[rightMotor] = 0;
+    }
+}
+
+void updateClaw()
+{
+    //TODO
+    if(vexRT[Btn5U] == 1)
+    {
+        motor[clawMotor] = 40;
+    }
+    //should tighten grabber
+    else if(vexRT[Btn5D] == 1)
+    {
+        motor[clawMotor] = -40;
+    }
+    // should loosen grabber
+    else
+        // should leave it tightened when button is relesed
+    {
+        motor[clawMotor] = 0;
+    }
+}
+
+void updateArm()
+{
+    //TOfinish
+    const int threshold = 10;
+
+    int joy1_y2 = vexRT[Ch3];
+
+    if(abs(joy1_y2) > threshold)
+    {
+        motor[armMotor] = vexRT[Ch3];     //modeled on dual joystick control and dual joystick control with arm
+    }
+
+}
+
+bool launchToggle = false;
+int lastBtn6dState = 0;
+
+void updateLaunch()
+{
+    int currentBtnState = vexRT[Btn6D];
+
+    // Only launch when last state was not pressed and current state is pressed
+    if(currentBtnState != lastBtn6dState && currentBtnState)
+    {
+        motor[launchServo] = launchToggle ? 100 : -100;

```

Comment:
Should be 128: -128

Reply:
The range here is, indeed, 128:-128. However, we don't need the full power of the motor here, we just need to be able to move a lever enough to release the catapult.

Reply:
The range here is, indeed, 128:-128. However, we don't need the full power of the motor here, we just need to be able to move a lever enough to release the catapult.